### PR TITLE
Clang format

### PR DESCRIPTION
**Pull request by lukasm91**
## Technical Description

- Adds a pre-commit hook that runs clang-format on any changed files
- Removes "make format" because this yoda-target does not what we want it should do
- Adds a `clang-format_all.sh` script in `scripts` (could be turned into a cmake target later)
- `storage_dsl.hpp` is the only file that is not only reformatted because comments were not handled properly by clang-format.

### Testing / Notes to the reviewer

- Run the clang-format script with your version (6.0!) and check that nothing changes
- Check the files which have been changed whether they are internal or external (i.e. check ignore list)

### CODE REVIEW

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

**Comment by lukasm91** [comment]
launch jenkins

File: scripts/git_hooks/pre-commit [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+# if false every changed file in the commit will be parsed with clang-format.
+# if true only files matching one of the extensions are parsed with clang-format.
+# PARSE_EXTS=true
+PARSE_EXTS=true
+
+# file types to parse. Only effective when PARSE_EXTS is true.
+# FILE_EXTS=".c .h .cpp .hpp"
+FILE_EXTS=".c .h .cpp .hpp .cc .hh .cxx .cu .m"
+
+source ${SCRIPTPATH}/../ignore_list.sh
```
Comment (havogt) [inline scripts/git_hooks/pre-commit]: Because of this line we need a symlink! copy doesn't work.

**Pull request merged**